core/include/cubos/core/ecs/world.hpp

```diff
@@ -27,6 +28,12 @@ namespace cubos::core::ecs
     class World
     {
     public:
+        /// @brief World constructor.
+        World()
+        {
+            this->registerComponent<Name>();
+        }
```

Comment:
I would move the definition of this to the the source file, thus removing the need to have the name.hpp include in the header.